### PR TITLE
Create part of tests for house member document service

### LINKED ISSUE #82: Compress images on the fly which are uploaded
## Compress the images uploaded to the document endpoint without staging

* As of now, the document/image which is uploaded is staged and then compressed and stored to DB.

## 💡 Describe the solution you'd like

* This should be done on the fly without staging.

### LINKED ISSUE #84: Create part of tests for HouseMemberDocument service (create and update document methods)
## ⚠️ Is your feature request related to a problem? Please describe

Create rest of unit tests for HouseMemberDocument service 

## 💡 Describe the solution you'd like

Create separate test file with applicationContext load

## 🤚 Do you want to develop this feature yourself?

- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by DDmit04**
## 🚀 Description

add more tests for `HouseMemberDocumentService`

## 📄 Motivation and Context

Closes #84 

## 📦 Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by DDmit04** [comment]
But I have a problem here I don't know how create image of a certain size (based on max filesize in `application.yaml`) for test case when uploaded image too large (even after compressing) so two of new tests not passed.

File: src/main/java/com/myhome/bootstrap/DataLoader.java [hunk 1]
```diff
@@ -59,7 +59,7 @@
 
   @Override public void run(String... args) throws Exception {
     //loadPostmanData();
-    loadData();
+    //loadData();
```
Comment (jmprathab) [inline src/main/java/com/myhome/bootstrap/DataLoader.java]: Why is this commented out?

File: build.gradle [hunk 2]
```diff
@@ -49,13 +49,18 @@ ext {
...
+
+
   // Spring web
   implementation 'org.springframework.boot:spring-boot-starter-web'
 
   // Spring test
   testImplementation('org.springframework.boot:spring-boot-starter-test') {
     exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
   }
+  testImplementation 'junit:junit:4.12'
```
Comment (jmprathab) [inline build.gradle]: Spring boot already includes JUnit. This is redundant, Can this be removed?

File: src/test/resources/application.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
+#
```
Comment (jmprathab) [inline src/test/resources/application.yml]: What is the need to duplicate the whole file? Can only required properties by added to this file?

File: src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+import java.io.IOException;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@SpringBootTest
```
Comment (jmprathab) [inline src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java]: Isn't this a unit test. Why is entire context necessary? I see all the dependencies being mocked.
In that case, can this annotation be removed.

Also, it is appropriate if this file is moved to `unit` folder.

**Review by jmprathab — CHANGES_REQUESTED** [review]

**Comment by jmprathab** [comment]
@DDmit04 CI seems to fail with `java.lang.NullPointerException at HouseMemberDocumentServiceTestAdditional.java:170`

File: src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    @Test
+    void updateHouseMemberDocumentSuccess() throws IOException {
+        // given
+        byte[] imageBytes = getImageAsByteArray(10, 10);
+        MockMultipartFile newDocumentFile = new MockMultipartFile("new-test-file-name", imageBytes);
+        HouseMember testMember = new HouseMember(MEMBER_ID, MEMBER_DOCUMENT, MEMBER_NAME, null);
+
+        given(houseMemberRepository.findByMemberId(MEMBER_ID))
+                .willReturn(Optional.of(testMember));
+        given(houseMemberDocumentRepository.save(any()))
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java]: Let's not use `any() ` but specific object

File: src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    void updateHouseMemberDocumentMemberNotExists() throws IOException {
+        // given
+        byte[] imageBytes = getImageAsByteArray(10, 10);
+        MockMultipartFile newDocumentFile = new MockMultipartFile("new-test-file-name", imageBytes);
+
+        given(houseMemberRepository.findByMemberId(MEMBER_ID))
+                .willReturn(Optional.empty());
+        // when
+        Optional<HouseMemberDocument> houseMemberDocument = houseMemberDocumentService.updateHouseMemberDocument(newDocumentFile, MEMBER_ID);
+        // then
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java]: Let's add a blank like before `then` section

File: src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+        MockMultipartFile newDocumentFile = new MockMultipartFile("new-test-file-name", imageBytes);
+        HouseMember testMember = new HouseMember(MEMBER_ID, MEMBER_DOCUMENT, MEMBER_NAME, null);
+
+        given(houseMemberRepository.findByMemberId(MEMBER_ID))
+                .willReturn(Optional.of(testMember));
+        given(houseMemberDocumentRepository.save(any()))
+                .willReturn(new HouseMemberDocument("mock-document-name", imageBytes));
+        // when
+        Optional<HouseMemberDocument> houseMemberDocument = houseMemberDocumentService.updateHouseMemberDocument(newDocumentFile, MEMBER_ID);
+        // then
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java]: Lets add empty line before `then`

**Review by mslowiak — CHANGES_REQUESTED** [review]

File: src/test/resources/application.yml (hunk 3 above)
Comment (DDmit04) [reply src/test/resources/application.yml]: I can remove this file

File: build.gradle (hunk 2 above)
Comment (DDmit04) [reply build.gradle]: yes

File: src/main/java/com/myhome/bootstrap/DataLoader.java (hunk 1 above)
Comment (DDmit04) [reply src/main/java/com/myhome/bootstrap/DataLoader.java]: it fails test because I use `Mockito.any()` now I fixed this

**Comment by DDmit04** [comment]
Also I fix issue #82 in 552f99e

File: build.gradle [hunk 8]
```diff
@@ -49,13 +49,18 @@ ext {
 }
 
 dependencies {
+
```
Comment (mslowiak) [inline build.gradle:52]: Unnecessary blank lines

File: build.gradle (hunk 2 above)
Comment (mslowiak) [inline build.gradle]: The JUnit is still in pom. 
Spring version contains JUnit 5 in `org.springframework.boot:spring-boot-starter-test` artifact that is included in our pom

File: src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.mock.web.MockMultipartFile;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java]: Can you fix that wildcard import?

File: src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+    
+    @Autowired
+    private HouseMemberDocumentSDJpaService houseMemberDocumentService;
+
+    @BeforeEach
+    private void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    private byte[] getImageAsByteArray(int height, int width) throws IOException {
```
Comment (mslowiak) [inline src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java]: Could you move this method to the end of this class?
Test classes should have tests and then 'helpers' methods

**Review by mslowiak — CHANGES_REQUESTED** [review]

File: src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java (hunk 4 above)
Comment (DDmit04) [reply src/test/java/com/myhome/services/HouseMemberDocumentServiceTestAdditional.java]: Fixed

**Pull request merged**